server/src/test/scala/org/http4s/server/middleware/DateSpec.scala

```diff
@@ -0,0 +1,74 @@
+package org.http4s.server.middleware
+
+import cats.implicits._
+import cats.effect._
+import cats.data.OptionT
+import org.http4s._
+import org.http4s.headers.{Date => HDate}
+import cats.effect.testing.specs2.CatsIO
+
+class DateSpec extends Http4sSpec with CatsIO {
+
+  implicit val T = IO.timer(scala.concurrent.ExecutionContext.global)
```

Comment:
Http4sSpec has an implicit `Timer[IO]` that doesn't use global, which leaks threads in long-running sbt sessions.